cmd/controller-manager/app/controller_manager.go

```diff
@@ -90,19 +90,20 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 	// Build the K8s kubeconfig / client / clientBuilder
 	glog.V(4).Info("Building k8s kubeconfig")
 
-	k8sKubeconfig, err := rest.InClusterConfig()
+	var err error
+	var k8sKubeconfig *rest.Config
+	if controllerManagerOptions.K8sAPIServerURL == "" && controllerManagerOptions.K8sKubeconfigPath == "" {
```

Comment:
> The BuildConfigFromFlags implicitly falls back to InClusterConfig, but gives some warning about how it might not work. By explicitly using InClusterConfig if we're missing both options, we can give better information in the logs so we don't have confusing log messages.

We don't provide a warning here.  Are we just checking this condition for consistency with how we setup the other config, or is there another reason?